### DISCUSSION TITLE
What if i need to use decimal percentages on the rarities?

### DISCUSSION THREAD

**Discussion by pietrociatta** [nftchef/art-engine#242] [Help + Q&A]
I'm trying to put like ````Background#5.5```` value, by changing to ```` const exp = /#(\d*\.\d{0,2})/; ````
But it seems not working, how can i use decimal value as rarities, also ````0.3```` for example.

**Comment by nftchef** [comment]
@pietrociatta thanks for the question, this comes up a lot.

Decimals are not supported in file name's rarity weights. 
If you need decimal percent rarity, you can calculate them by using a higher total weight number. 

reminder, `weight`'s in a folder **do not need to add up to 100**, it can be anything. Percentage at the time of generation is calculated as
```
percent change = file weight / total sum of weights in folder
```
to have a "decimal percent chance" for a given file, you can have the total weights add up to 1000. 
Then, `some-file#5.png` would have a `0.5%` chance of being chosen

**Reply by null-prophet** [reply]
you really should put this in the readme, I had to look through all the code and figure this out when it could have simply been said in the readme that this is how it works.
